app/src/main/java/org/osservatorionessuno/bugbane/screens/ScanScreen.kt

```diff
@@ -361,12 +362,13 @@ fun ScanScreen(
                     )
                     Spacer(modifier = Modifier.width(8.dp))
                     Text(
-                        text = if (isScanning)
-                            stringResource(R.string.home_scanning_button)
-                        else if (lacksPermissions)
-                            stringResource(R.string.home_permissions_button)
-                        else
-                            stringResource(R.string.home_scan_button),
+                        text = when (appState.value) {
+                            AppState.AdbScanning -> stringResource(R.string.home_scanning_button)
+                            AppState.AdbConnected -> stringResource(R.string.home_scan_button)
+                            AppState.TryAutoConnect, AppState.AdbConnecting -> stringResource(R.string.notification_adb_pairing_working_title)
```

Comment:
(as with above, `notification_adb_pairing_working_title` could be improved on here to ~ "trying to connect to adb" but just tried to reuse existing strings and avoid more translation burden until things are more finalized)